sunpy/net/jsoc/jsoc.py

```diff
@@ -483,189 +639,166 @@ def get_request(self, requestIDs, path=None, overwrite=False, progress=True,
-            # pragma: no cover
+        if 'series' not in iargs:
+            error_message = "Series must be specified for a JSOC Query"
+            raise ValueError(error_message)
 
+        if not isinstance(keywords, list) and not isinstance(keywords, six.string_types):
+            error_message = "Keywords can only be passed as a list or "\
+                            "comma-separated strings."
+            raise TypeError(error_message)
+
+        # Raise errors for PrimeKeys
+        pkeys = c.pkeys(iargs['series'])
+        pkeys_passed = iargs.get('primekey', None)
+        if pkeys_passed is not None:
+            if not set(list(pkeys_passed.keys())) <= set(pkeys):
+                error_message = "Unexpected PrimeKeys were passed. The series {series} "\
+                                "supports the following PrimeKeys {pkeys}"
+                raise ValueError(error_message.format(series=iargs['series'], pkeys=pkeys))
+
+        # Raise errors for wavelength
+        wavelength = iargs.get('wavelength', '')
+        if wavelength:
+            if 'WAVELNTH' not in pkeys:
+                error_message = "The series {series} does not support wavelength attribute."\
+                                " Following primekeys are supported {pkeys}"
+                raise TypeError(error_message.format(series=iargs['series'], pkeys=pkeys))
+
+        # Raise errors for segments
+        si = c.info(iargs['series'])
+        segs = list(si.segments.index.values)
+        segs_passed = iargs.get('segment', '')
+        if segs_passed:
+            if not isinstance(segs_passed, list):
+                error_message = "Segments can only be passed as a list of strings."
+                raise TypeError(error_message)
+
+            if not set(segs_passed) <= set(segs):
+                error_message = "Unexpected Segments were passed. The series {series} "\
+                                "contains the following Segments {segs}"
+                raise ValueError(error_message.format(series=iargs['series'], segs=segs))
+
+        if 'start_time' in iargs:
+            iargs['start_time'] = iargs['start_time'].tai.datetime
+            iargs['end_time'] = iargs['end_time'].tai.datetime
+
+        ds = self._make_recordset(**iargs)
+        if isinstance(keywords, list):
+            key = str(keywords)[1:-1].replace(' ', '').replace("'", '')
         else:
-            returns.append(response)
+            key = keywords
 
-        return returns
+        r = c.query(ds, key=key, rec_index=isMeta)
 
-    def _request_status(self, request_id):
-        """
-        GET the status of a request ID
-        """
-        payload = {'op': 'exp_status', 'requestid': request_id}
-        u = requests.get(JSOC_EXPORT_URL, params=payload)
+        if isMeta:
+            return r
 
-        return u
+        if r is None or r.empty:
+            return astropy.table.Table()
```

Comment:
This, and below should be `QTable` just as a matter of neatness.

Reply:
Did this change not need to be made?